File: Java codes/QueueUsingArray.java
import java.io.*;

class Queue
{
	private int[] queue;
	private int capacity;
	private int size;
	private int front;
	private int rear;

	Queue()
	{
		this.queue = new int[2];
		this.capacity = 2;
		this.size = 0;
		this.front = -1;
		this.rear = -1;
	}

	public void enqueue(int val)
	{
		if (size == capacity)
		{
			int[] temp = new int[capacity*2];
			int j = 0;
			for (int i = front; i != rear; i = (i+1)%capacity)
				temp[j++] = queue[i];
			temp[j] = queue[rear];
			front = 0;
			rear = j;
			capacity *= 2;
			queue = temp;
		}
		if (size == 0)
			front = rear = 0;
		else
			rear = (rear+1)%capacity;
		queue[rear] = val;
		++size;
	}

	public int dequeue()
	{
		int val = queue[front];
		if (size == 1)
			front = rear = -1;
		else
			front = (front+1)%capacity;
		--size;
		return val;
	}

	public int checkFront()
	{
		return queue[front];
	}

	public boolean isEmpty()
	{
		if (front == rear && front == -1)
			return true;
		return false;
	}
}

class QueueUsingArray
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		Queue queue = new Queue();
		while (true)
		{
			System.out.println("1- Enqueue\n2- Dequeue\n3- Check Front\n4- Check if empty\n5- Quit");
			int choice = Integer.parseInt(br.readLine());
			switch(choice)
			{
				case 1:
					System.out.println("Value- ");
					queue.enqueue(Integer.parseInt(br.readLine()));
					break;

				case 2:
					if (queue.isEmpty())
						System.out.println("Empty Queue");
					else
						System.out.println(queue.dequeue()+" removed");
					break;

				case 3:
					if (queue.isEmpty())
						System.out.println("Empty Queue");
					else
						System.out.println(queue.checkFront()+" is at the front");
					break;

				case 4:
					if (queue.isEmpty())
						System.out.println("Empty");
					else
						System.out.println("Not empty");
					break;

				case 5:
					System.out.println("Quit");
					System.exit(0);

				default:
					System.out.println("Invalid Input");
			}
		}
	}
}

File: C codes/QueueUsingArray.c
#include <stdio.h>
#include <stdlib.h>

int num[10];
int SIZE=10, front=-1, rear=-1;

void enqueue();
void dequeue();
void display();

void enqueue()
{
	if(rear==SIZE-1)
		printf("Queue overflow\n");
	else if(front==-1&&rear==-1)
	{
		front=rear=0;
		printf("Enter the information-\n");
		scanf("%d",&num[rear]);
		display();
	}
	else
	{
		++rear;
		printf("Enter the information-\n");
		scanf("%d",&num[rear]);
		display();
	}
}

void dequeue()
{
	if(front==-1&&rear==-1)
		printf("Queue underflow\n");
	else if(front==rear)
	{
		printf("%d is deleted\n",num[front]);
		front=rear=-1;
		display();
	}
	else
	{
		printf("%d is deleted\n",num[front]);
		++front;
		display();
	}
}

void display()
{
	int i;
	if(front==-1&&rear==-1)
		printf("Queue is empty\n");
	else
	{
		printf("The queue is-\n");
		for(i=front;i<=rear;i++)
			printf("%d\t",num[i]);
	}
}

void main()
{
	printf("\t\t\tQUEUE\n");
	printf("Queue size is 10\n");
	while(1)
	{
		int choice;
		printf("\n\t\t\tSEE THE OPTIONS\n\t\t\t1- Enqueue\n\t\t\t2- Dequeue\n\t\t\t3- Display\n\t\t\t4- Quit\n\t\t\tEnter your choice-\n");
		scanf("%d",&choice);
		switch(choice)
		{
			case 1:
				enqueue();
				break;
			case 2:
				dequeue();
				break;
			case 3:
				display();
				break;
			case 4:
				printf("\t\t\tTHANKS FOR USING MY APPLICATION!!!");
				return;
				break;
			default:
				printf("INVALID INPUT\n");
				break;
		}
	}
}